server/src/test/java/keywhiz/service/permission/PermissionCheckTest.java

```diff
@@ -0,0 +1,51 @@
+package keywhiz.service.permission;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.inject.Provider;
+import keywhiz.service.permissions.Action;
+import keywhiz.service.permissions.AlwaysAllowPermissionCheck;
+import keywhiz.service.permissions.AlwaysFailPermissionCheck;
+import keywhiz.service.permissions.KeywhizPrincipal;
+import keywhiz.service.permissions.PermissionCheck;
+import keywhiz.service.permissions.PermissionCheckImpl;
+import keywhiz.service.permissions.PermissionCheckProvider;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PermissionCheckTest {
+
+  @Rule public MockitoRule mockito = MockitoJUnit.rule();
+
+  @Mock MetricRegistry metricRegistry;
+  @Mock KeywhizPrincipal principal;
+  @Mock Object target;
+  Provider<PermissionCheck> permissionCheckProvider;
+  PermissionCheckImpl permissionCheck;
+  String add = Action.ADD;
+  String create = Action.CREATE;
+  String delete = Action.DELETE;
+
+  @Before
+  public void setUp() {
+    permissionCheckProvider = new PermissionCheckProvider(metricRegistry);
+    permissionCheck = new PermissionCheckImpl(permissionCheckProvider);
+  }
+
+  @Test public void IsAllowedAlwaysReturnsTrue() {
+    boolean permitted_add = permissionCheck.isAllowed(principal, add, target);
+    assertThat(permitted_add).isEqualTo(true);
+
+    boolean permitted_create = permissionCheck.isAllowed(principal, create, target);
+    assertThat(permitted_create).isEqualTo(true);
+
+    boolean permitted_delete = permissionCheck.isAllowed(principal, delete, target);
+    assertThat(permitted_delete).isEqualTo(true);
+  }
```

Comment:
Make sure you follow the variable naming convention for Java (camelCase).

As far as how to implement this test, the behavior you want to verify is that the permission check always returns true regardless of the result of the underlying permission check (which could be true or false, or it could throw an exception). So you'd want to have a test for each of those conditions, and the easiest way to do that is to use a @ Mock PermissionCheck as the delegate to the AlwaysAllowPermissionCheck and then control what the output of the mock is.

I like to think of tests in terms of truth tables, so something like:

* Delegate: T Result: T
* Delegate: F Result: T
* Delegate: X Result: T

(where X is exception, although I don't know if we actually discussed what we should do in that case)

Basically, what are all the possible combinations of inputs and expected outputs, and is there a test for each combination?